libtenzir/src/execution_node.cpp

```diff
@@ -746,6 +746,23 @@ auto exec_node(
     return exec_node_actor::behavior_type::make_empty_behavior();
   }
   self->state.weak_node = node;
+  self->set_exception_handler(
+    [self](std::exception_ptr exception) -> caf::error {
+      try {
+        std::rethrow_exception(exception);
+      } catch (diagnostic diag) {
+        self->state.ctrl->diagnostics().emit(std::move(diag));
+        return {};
+      } catch (const std::exception& err) {
+        diagnostic::error("{}", err.what())
+          .note("unhandled exception in {} {}", *self, self->state.op->name())
+          .emit(self->state.ctrl->diagnostics());
+        return {};
+      }
+      return diagnostic::error("unhandled exception in {} {}", *self,
+                               self->state.op->name())
+        .to_error();
+    });
```

Comment:
@Dakostu there really isn't a good way to test this right now, as usually we would consider an uncaught exception a bug. I verified that this works manually by modifying an operator to throw one.